### PR TITLE
Fixing Rspec 3.2 update.  We were overriding private APIs which changed.

### PR DESCRIPTION
**Pull request by tyler-ball**
This should fix the recent Travis errors we were seeing, and overwrites https://github.com/chef/chef/pull/2854

### CODE REVIEW

**Comment by jdmundrawala** [comment]
LGTM

**Comment by tyler-ball** [comment]
\cc @chef/client-core

File: chef.gemspec [hunk 1]
```diff
@@ -36,7 +36,7 @@ Gem::Specification.new do |s|
   s.add_dependency 'plist', '~> 3.1.0'
 
   # Audit mode requires these, so they are non-developmental dependencies now
-  %w(rspec-core rspec-expectations rspec-mocks).each { |gem| s.add_dependency gem, "~> 3.1.0" }
+  %w(rspec-core rspec-expectations rspec-mocks).each { |gem| s.add_dependency gem, "~> 3.2.0" }
```
Comment (lamont-granquist) [inline chef.gemspec]: you should be able to go to `~> 3.2` since we're using public apis that shouldn't change in 3.3

**Comment by lamont-granquist** [comment]
:+1:

**Comment by danielsdeleo** [comment]
I'm with Lamont, we should be able to pin to ~> 3.2

LGTM otherwise

**Comment by mcquin** [comment]
Woo! :+1:

**Comment by jdmundrawala** [comment]
:+1: on the change to rspec btw

**Pull request merged**